orca-queue-redis/src/main/kotlin/com/netflix/spinnaker/orca/q/redis/RedisQueue.kt

```diff
@@ -83,7 +82,17 @@ class RedisQueue(
         ?.also { id ->
           val ack = this::ackMessage.partially1(id)
           redis.readMessage(id) { message ->
-            callback(message, ack)
+            val attempts = message.getAttribute(AttemptsAttribute::class.java)!!.attempts
+            val maxAttempts = message.getAttribute(MaxAttemptsAttribute::class.java)?.maxAttempts ?: 0
+
+            if (maxAttempts > 0 && attempts > maxAttempts) {
+              log.warn("Message $id with payload $message exceeded $maxAttempts retries")
+              handleDeadMessage(message)
```

Comment:
Borrowed this handling (`handleDeadMessage()`, etc.) from the other place we checked maxAttempts.

Not really sure it needs to be DLQ'd ... it's also possible we might need to send another message to ensure a reasonable exception message gets into the `Stage` context (so that `deck` could inform an end user what happened).